backend/remote-state/pg/backend.go

```diff
@@ -64,9 +71,25 @@ func (b *Backend) configure(ctx context.Context) error {
 
 	// Prepare database schema, tables, & indexes.
 	var query string
-	query = `CREATE SCHEMA IF NOT EXISTS %s`
-	if _, err := db.Exec(fmt.Sprintf(query, b.schemaName)); err != nil {
-		return err
+
+	if !data.Get("skip_schema_creation").(bool) {
+		// list all schemas to see if it exists
+		var count int
+		query = `select count(1) from information_schema.schemata where lower(schema_name) = lower('%s')`
+		if err := db.QueryRow(fmt.Sprintf(query, b.schemaName)).Scan(&count); err != nil {
+			return err
+		}
+
+		// skip schema creation if schema already exists
+		// `CREATE SCHEMA IF NOT EXISTS` is to be avoided if ever
+		// a user hasn't been granted the `CREATE SCHEMA` privilege
+		if count < 1 {
+			// tries to create the schema
+			query = `CREATE SCHEMA %s`
```

Comment:
While logically the schema was just queried for existence, these queries are not in an isolated transaction together, so this `CREATE SCHEMA` command could fail if it races with another client doing the same thing.

Please keep the `IF NOT EXISTS` in these creation queries to avoid such errors:

```suggestion
			query = `CREATE SCHEMA IF NOT EXISTS %s`
```